### ISSUE TITLE
overloaded functon renaming might cause conflicts

### ISSUE THREAD

**Issue by hfiguiere** [google/autocxx#1316]
**Describe the bug**

overloaded functon renaming might cause conflicts

**To Reproduce**

I have  a class:

```c++
class Image {
public:
  static uint64_t byteSwap(uint64_t value, bool bSwap);
  static uint32_t byteSwap(uint32_t value, bool bSwap);
  static uint16_t byteSwap(uint16_t value, bool bSwap);
  static uint16_t byteSwap2(const DataBuf& buf, size_t offset, bool bSwap);
};
```
calling `generate!` on it will cause `byteSwap2` to be generated 2.
```rust
pub unsafe fn byteSwap(value: u64, bSwap: bool) -> u64 { }
pub unsafe fn byteSwap1(value: u32, bSwap: bool) -> u32 { }
pub unsafe fn byteSwap2(value: u16, bSwap: bool) -> u16 { }
pub unsafe fn byteSwap2(buf: &root::Exiv2::DataBuf,
       offset: usize,
       bSwap: bool,
) -> u16 {}
```

This is because the renaming of the return u16 overload is `byteSwap2` and there is an actual `byteSwap2` function.

**Expected behavior**

Find a better renaming.

Or throw an error from the macro

**Additional context**

**Comment by adetaylor** [comment]
Yeah, this isn't great and [as the manual says, this all needs to be overhauled a bit](https://google.github.io/autocxx/cpp_functions.html#overloads---and-identifiers-ending-in-digits). Meanwhile though I do certainly consider this a bug, and I'd accept a pull request for a temporary solution, e.g. if `SomeOverloadedFunction2` already exists, then use a different suffix such as `SomeOverloadedFunction_autocxx_overload2`. I'm unlikely to get a chance to do this any time soon, but this might be a fairly easy fix so I'm going to mark it as Good First Issue in case someone else can help out.

**Comment by adetaylor** [comment]
Closely related to #995

**Comment by minghuaw** [comment]
Just sharing some random thoughts on overloaded functions. Would the following options be better?

1. Generate macros for the overloaded functions?
    
    A potential problem I can see is overloads that have the same number of args but the args are of different types, like the three `byteSwap` functions mentioned above.

2. Generate trait for the overloaded functions?

    What I am picturing is something like below. I guess a potential problem would be generating multiple traits of the same name.

```rust
struct Image;

pub trait ByteSwap<Args> {
    type Output;

    fn byte_swap(args: Args) -> Self::Output;
}

impl ByteSwap<(u64, bool)> for Image {
    type Output = u64;

    fn byte_swap(args: (u64, bool)) -> u64 { todo!() }
}


impl ByteSwap<(u32, bool)> for Image {
    type Output = u32;

    fn byte_swap(args: (u32, bool)) -> u32 { todo!() }
}

impl ByteSwap<(u16, bool)> for Image {
    type Output = u16;

    fn byte_swap(args: (u16, bool)) -> u16 { todo!() }
}
```

**Comment by adetaylor** [comment]
This is an aspect of #520.

**Comment by SatvikMishra08** [comment]
Claiming #1316 — will make overload renames skip identifiers that already exist (e.g. byteSwap→byteSwap2 vs real byteSwap2) using `_autocxx_overload{N}` per maintainer guidance, plus the PR #1317 repro test.
